internal/character/hook/skill.go

```diff
@@ -0,0 +1,125 @@
+package hook
+
+import (
+	"github.com/simimpact/srsim/internal/global/common"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const Skill key.Attack = "hook-skill"
+const EnhancedSkill key.Attack = "hook-enhanced-skill"
+
+func (c *char) Skill(target key.TargetID, state info.ActionState) {
+	if c.engine.HasModifier(c.id, SkillEnhancement) {
+		c.EnhancedSkill(target)
+	} else {
+		c.NormalSkill(target)
+	}
+
+	c.engine.EndAttack()
+
+	c.applySkillBurn([]key.TargetID{target})
+}
+
+func (c *char) NormalSkill(target key.TargetID) {
+	c.engine.Attack(info.Attack{
+		Key:        Skill,
+		Targets:    []key.TargetID{target},
+		Source:     c.id,
+		AttackType: model.AttackType_SKILL,
+		DamageType: model.DamageType_FIRE,
+		BaseDamage: info.DamageMap{
+			model.DamageFormula_BY_ATK: skillNormal[c.info.SkillLevelIndex()],
+		},
+		EnergyGain:   30,
+		StanceDamage: 60,
+	})
+
+	if c.engine.HasBehaviorFlag(target, model.BehaviorFlag_STAT_DOT_BURN) {
+		c.talentProc(target)
+	}
+}
+
+// Special checks to mimic dm/also to avoid multiple procs of the energy gain
+func (c *char) EnhancedSkill(target key.TargetID) {
+	// Main target
+	c.engine.Attack(info.Attack{
+		Key:        EnhancedSkill,
+		Targets:    []key.TargetID{target},
+		Source:     c.id,
+		AttackType: model.AttackType_SKILL,
+		DamageType: model.DamageType_FIRE,
+		BaseDamage: info.DamageMap{
+			model.DamageFormula_BY_ATK: skillEnhanceMain[c.info.SkillLevelIndex()],
+		},
+		EnergyGain:   30,
+		StanceDamage: 60,
+	})
+
+	// Adjacent targets
+	c.engine.Attack(info.Attack{
+		Key:        EnhancedSkill,
+		Targets:    c.engine.AdjacentTo(target),
+		Source:     c.id,
+		AttackType: model.AttackType_SKILL,
+		DamageType: model.DamageType_FIRE,
+		BaseDamage: info.DamageMap{
+			model.DamageFormula_BY_ATK: skillEnhanceAdj[c.info.SkillLevelIndex()],
+		},
+		EnergyGain:   0,
+		StanceDamage: 30,
+	})
+
+	talentCanidates := c.engine.Retarget(info.Retarget{
+		Targets: append(c.engine.AdjacentTo(target), target),
+		Filter: func(target key.TargetID) bool {
+			return c.engine.HasBehaviorFlag(target, model.BehaviorFlag_STAT_DOT_BURN)
+		},
+		IncludeLimbo: true,
+	})
+
+	// Cannot simply loop over every enemy from the retarget with talentProc
+	// since that would trigger hp restore and energy restore multiple times
+	if len(talentCanidates) > 0 {
+		c.engine.ModifyEnergy(info.ModifyAttribute{
+			Key:    Talent,
+			Target: c.id,
+			Source: c.id,
+			Amount: 5,
+		})
+		for _, t := range talentCanidates {
+			c.talentPursuedDamage(t)
+		}
+		c.applySkillBurn(c.engine.AdjacentTo(target))
+		c.talentHeal()
+	}
```

Comment:
next is here in an else. If there are no talent candidates, last hit is the attacks on adjacent targets
```suggestion
	} else {
	    c.engine.EndAttack()
	}
```
...and that's it i think. i see you've already added the one in `talentPursuedDamage`.